### PR TITLE
RMB-253: Metadata without user id

### PR DESCRIPTION
**Pull request by julianladisch**

### CODE REVIEW

File: domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+
+-- Restrictions specified in metadata.schema:
+-- jsonb->'metadata' is optional, but if it exists then jsonb->'metadata'->>'createdDate' is required.
+
+-- Trigger for insert: Copy createdDate and createdByUserId to creation_date and created_by.
 CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.${table.tableName}_set_md()
 RETURNS TRIGGER AS $$
 BEGIN
-  NEW.creation_date = to_timestamp(NEW.jsonb->'metadata'->>'createdDate', 'YYYY-MM-DD"T"HH24:MI:SS.MS');
+  NEW.creation_date = NEW.jsonb->'metadata'->>'createdDate' || '+0000';
```
Comment (opochepam) [inline domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl]:
```suggestion
  NEW.creation_date = (NEW.jsonb->'metadata'->>'createdDate')::timestamp AT TIME ZONE 'UTC';
```

File: domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl [hunk 2]
```diff
@@ -16,39 +33,22 @@ DROP TRIGGER IF EXISTS set_${table.tableName}_md_trigger ON ${myuniversity}_${my
...
+    RETURN NEW;
   end if;
-  if createdDate IS NOT NULL then
--- creation date and update date will always be injected by rmb - if created date is null it means that there is no meta data object
--- associated with this object - so only add the meta data if created date is not null -- created date being null may be a problem
--- and should be handled at the app layer for now -- currently this protects against an exception in the db if no md is present in the json
-    injectedId = '{"createdDate":"'||to_char(createdDate,'YYYY-MM-DD"T"HH24:MI:SS.MS')||'" , "createdByUserId":"'||createdBy||'", "updatedDate":"'||to_char(updatedDate,'YYYY-MM-DD"T"HH24:MI:SS.MSOF')||'" , "updatedByUserId":"'||updatedBy||'"}';
-    NEW.jsonb = jsonb_set(NEW.jsonb, '{metadata}' ,  injectedId::jsonb , false);
+
+  NEW.jsonb = jsonb_set(NEW.jsonb, '{metadata,createdDate}', to_jsonb(NEW.creation_date AT TIME ZONE '+0000'));
```
Comment (opochepam) [inline domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl]: seems like there is no need to specify timezone here as creation_date is already saved at 'UTC'

**Comment by julianladisch** [comment]
Please re-review the fixes.

**Comment by adamdickmeiss** [comment]
Please squash and merge.. The headline should be RMB-353 (not RMB-253)

**Comment by julianladisch** [comment]
Closed in favor of https://github.com/folio-org/raml-module-builder/pull/414

**Pull request closed without merging**